static/app/views/insights/mobile/screens/components/screensOverview.tsx

```diff
@@ -0,0 +1,235 @@
+  const primaryFields = isSpanPrimary ? spanMetricsFields : transactionMetricsFields;
+
+  const secondaryDataset = isSpanPrimary ? transactionMetricsDataset : spanMetricsDataset;
+  const secondaryFields = isSpanPrimary ? transactionMetricsFields : spanMetricsFields;
+
+  const {
+    data: primaryData,
+    isLoading: primaryLoading,
+    pageLinks: primaryLinks,
+    eventView: primaryEventView,
+  } = useMetrics(
+    primaryDataset,
+    primaryFields,
+    isProjectCrossPlatform ? selectedPlatform : undefined,
+    selection,
+    location,
+    sortedBy,
+    true,
+    []
+  );
+
+  const {
+    data: secondaryData,
+    // isLoading: secondaryLoading,
+    // pageLinks: secondaryLinks,
+    // eventView: secondaryEventView,
+  } = useMetrics(
+    secondaryDataset,
+    secondaryFields,
+    isProjectCrossPlatform ? selectedPlatform : undefined,
+    selection,
+    location,
+    undefined,
+    visibleScreens.length > 0,
+    visibleScreens
+  );
+
+  useEffect(() => {
+    if (primaryData) {
+      const screens: string[] = new Array();
+      primaryData?.data.forEach(row => {
+        if (row.transaction) {
+          screens.push(String(row.transaction));
+        }
+      });
+      setVisibleScreens(screens);
+    }
+  }, [primaryData]);
+
+  const tableSearchFilters = new MutableSearch(['transaction.op:ui.load']);
+  const derivedQuery = getTransactionSearchQuery(location, primaryEventView.query);
+
+  const combinedData = useMemo((): TableData | undefined => {
+    if (primaryData !== undefined && secondaryData !== undefined) {
+      const meta: MetaType = {};
+      meta.units = {
+        ...secondaryData.meta?.units,
+        ...primaryData.meta?.units,
+      };
+      meta.fields = {
+        ...secondaryData.meta?.fields,
+        ...primaryData.meta?.fields,
+      };
+
+      const data = primaryData.data.map(row => {
+        const matchingRow = secondaryData.data.find(
+          metricRow => metricRow.transaction === row.transaction
+        );
+        if (matchingRow) {
+          return {
+            ...matchingRow,
+            ...row,
+          };
+        }
+        return row;
+      });
+
+      return {
+        data: data,
+        meta: meta,
+      };
+    }
+
+    return primaryData;
+  }, [primaryData, secondaryData]);
+
+  const loading = primaryLoading;
+
+  return (
+    <Container>
+      <SearchBar
+        eventView={primaryEventView}
+        onSearch={search => {
+          router.push({
+            pathname: router.location.pathname,
+            query: {
+              ...location.query,
+              cursor: undefined,
+              query: String(search).trim() || undefined,
+            },
+          });
+        }}
+        organization={organization}
+        query={getFreeTextFromQuery(derivedQuery)}
+        placeholder={t('Search for Screen')}
+        additionalConditions={new MutableSearch(tableSearchFilters.formatString())}
```

Comment:
Is it possible to just pass along `tableSearchFilters` here? I think it's of type `MutableSearch` so we could avoid formatting it to a string to re-instantiate it